okonomiyaki/utils/misc.py

```diff
@@ -93,5 +100,51 @@ def _resolve(d):
     return ret
 
 
-def substitute_variable(v, local_vars):
-    return string.Template(v).substitute(local_vars)
+class RequireCurlyTemplate(string.Template):
+    """This class inheriting from Template requires curly braces.
+       A '$' without curly braces will not be substituted.
+    """
+    delimiter = '$'
+    # named and escaped groups are always None
+    # This is because their patterns are a subset of the invalid group,
+    # i.e. the invalid group will always match first.
+    # According to the Python re documentation the "|" operator is never greedy,
+    # so the named and escaped groups will always be None.
+    ignore_escape_pattern_str = r"""
+    (?<!\$)\$(?:                        # Only match single dollar signs
+      {(?P<braced>[_a-z][_a-z0-9]*)} |  # Delimiter and braced identifier
+      {(?P<invalid>[^}]*)}           |  # Other ill-formed delimiter expr
+      {(?P<named>)}                  |  # named group is always None
+      {(?P<escaped>)}                   # escaped group is always None
+    )
+    """
+    ignore_escape_pattern = re.compile(
+        ignore_escape_pattern_str, re.IGNORECASE | re.VERBOSE
+    )
+    pattern = r"""
+    \$(?:
+      (?P<escaped>\$)(?={[^}]*})     |  # Extra delimiter followed by braces
+      {(?P<braced>[_a-z][_a-z0-9]*)} |  # Delimiter and braced identifier
+      {(?P<invalid>[^}]*)}           |  # Other ill-formed delimiter expr
+      {(?P<named>)}                     # named group is always None
+    )
+    """
+
+    def __init__(self, template, ignore_escape=False):
+        super(RequireCurlyTemplate, self).__init__(template)
+        if ignore_escape:
+            self.pattern = self.ignore_escape_pattern
+
+
+def substitute_variable(v, local_vars,
+                        template='standard',
+                        ignore_escape=False):
```

Comment:
There is no need for this option.

Reply:
offline discussion with kevin proved me wrong. So we need to have this option so that we can control when escaping with take place.